ckanext/unaids/tests/test_dataset_transfer.py

```diff
@@ -1,16 +1,83 @@
-'Tests for plugin.py.'
 # encoding: utf-8
+import ckan.model as model
 from ckan.tests import helpers
 from ckan.lib.helpers import url_for
 from ckan.tests import factories
 import pytest
+import mock
 
+from ckanext.unaids.dataset_transfer.model import init_tables
+from ckanext.unaids.dataset_transfer.logic import (
+    _get_users_to_email,
+    send_dataset_transfer_emails
+)
 
+
+@pytest.fixture
+def initdb():
+    model.Session.remove()
+    model.Session.configure(bind=model.meta.engine)
+    init_tables()
+
+
+@pytest.mark.usefixtures('initdb')
 @pytest.mark.ckan_config('ckan.plugins', 'unaids scheming_datasets')
 @pytest.mark.usefixtures('with_plugins')
 @pytest.mark.usefixtures('clean_db')
 class TestDatasetTransfer(object):
 
+    @mock.patch('ckan.lib.mailer.mail_user')
+    def test_send_dataset_transfer_emails(self, mocked_mail_user, app):
+        user_1, user_2 = [
+            factories.User()
+            for x in range(2)
+        ]
+        org_1, org_2 = [
+            factories.Organization(user=user)
+            for user in [user_1, user_2]
+        ]
+        dataset = factories.Dataset(
+            owner_org=org_1['id'],
+            type='test-schema',
+            org_to_allow_transfer_to=org_2['id']
+        )
+        send_dataset_transfer_emails(
+            dataset_id=dataset['id'],
+            recipient_org_id=org_2['id']
+        )
+        mocked_mail_user.assert_called()
+        emailed_user_ids = [
+            x[0][0].id  # recipient_user_id
+            for x in mocked_mail_user.call_args_list
+        ]
+        assert emailed_user_ids == [user_2['id']]
+
+    def test_get_users_to_email(self, app):
```

Comment:
This is an all-in-one test case. Please create a separate test case for each of a use case of the function under test.